### PR TITLE
Port 4111 update terraform docs

### PR DESCRIPTION
**Pull request by dvirsegev**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

## Added docs pages

Please also include the path for the added docs

- Quickstart (`/`)
- Blueprint (`/platform-overview/port-components/blueprint`)
- ...

## Updated docs pages

Please also include the path for the updated docs

- Quickstart (`/`)
- Blueprint (`/platform-overview/port-components/blueprint`)
- ...

### CODE REVIEW

File: docs/build-your-software-catalog/sync-data-to-catalog/iac/terraform/terraform.md [hunk 1]
```diff
@@ -308,27 +297,39 @@ resource "port-labs_entity" "myEntity" {
...
 <TabItem value="many">
 
-The following parameters are **required**:
+the schema is as follows:
 
-- `name` - the `identifier` of the [relation](../../../define-your-data-model/relate-blueprints/relate-blueprints.md#structure-table) in the blueprint definition;
-- `identifiers` - the identifiers of the target entities.
+```hcl showLineNumbers
+relations {
+  many_relations = {
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/iac/terraform/terraform.md:319]: many_relation or many_relation**s**

File: docs/configuration-methods/configuration-methods.md [hunk 2]
```diff
@@ -62,10 +62,10 @@ Port's [Terraform provider](https://registry.terraform.io/providers/port-labs/po
 
 <InstallTerraform />
 
-To create a blueprint using Port's Terraform provider you need a `.tf` file defining a [`port-labs_blueprint`](https://registry.terraform.io/providers/port-labs/port-labs/latest/docs/resources/blueprint) resource:
+To create a blueprint using Port's Terraform provider you need a `.tf` file defining a [`port_blueprint`](https://registry.terraform.io/providers/port-labs/port-labs/latest/docs/resources/blueprint) resource:
 
 ```hcl showLineNumbers
-resource "port-labs_blueprint" "myBlueprint" {
+resource "port_blueprint" "myBlueprint" {
```
Comment (MPTG94) [inline docs/configuration-methods/configuration-methods.md:68]: Will the plan for this resource run properly without configuring any properties?

File: docs/create-self-service-experiences/setup-ui-for-action/setup-ui-for-action.md [hunk 3]
```diff
@@ -192,7 +192,7 @@ The JSON shown above is for a single blueprint action, the actions of a blueprin
 <TabItem value="terraform" label="Terraform">
 
 ```hcl showLineNumbers
-resource "port-labs_action" "myAction" {
+resource "port_action" "myAction" {
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-ui-for-action/setup-ui-for-action.md:195]: The user inputs syntax here needs to be updated

File: docs/create-self-service-experiences/setup-ui-for-action/user-inputs/array.md [hunk 4]
```diff
@@ -77,7 +77,7 @@ The array input type can be used to store any list of data, for example:
 <TabItem value="basic">
 
 ```hcl showLineNumbers
-resource "port-labs_action" "myAction" {
+resource "port_action" "myAction" {
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-ui-for-action/user-inputs/array.md:80]: The user inputs syntax here needs to be updated

File: docs/create-self-service-experiences/setup-ui-for-action/user-inputs/text.md [hunk 5]
```diff
@@ -99,7 +99,7 @@ In the [live demo](https://demo.getport.io/self-serve) self-service hub page, we
 <TabItem value="basic">
 
 ```hcl showLineNumbers
-resource "port-labs_action" "myAction" {
+resource "port_action" "myAction" {
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-ui-for-action/user-inputs/text.md:102]: The user inputs syntax here needs to be updated

Also if there is now support for enum and array, please extend the examples properly

File: docs/create-self-service-experiences/setup-ui-for-action/user-inputs/url.md [hunk 6]
```diff
@@ -99,7 +99,7 @@ The URL input type can be used to store a link to any web resource, for example:
 <TabItem value="basic">
 
 ```hcl showLineNumbers
-resource "port-labs_action" "myAction" {
+resource "port_action" "myAction" {
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-ui-for-action/user-inputs/url.md:102]: The user inputs syntax here needs to be updated

Also if there is now support for enum and array, please extend the examples properly

File: docs/create-self-service-experiences/setup-ui-for-action/user-inputs/user.md [hunk 7]
```diff
@@ -82,7 +82,7 @@ In the [live demo](https://demo.getport.io/self-serve) self-service hub page, we
 <TabItem value="basic">
 
 ```hcl showLineNumbers
-resource "port-labs_action" "myAction" {
+resource "port_action" "myAction" {
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-ui-for-action/user-inputs/user.md:85]: The user inputs syntax here needs to be updated

Also if there is now support for enum and array, please extend the examples properly

File: docs/create-self-service-experiences/setup-ui-for-action/user-inputs/yaml.md [hunk 8]
```diff
@@ -81,7 +81,7 @@ The yaml input type can be used to store any key/value based data, for example:
 <TabItem value="basic">
 
 ```hcl showLineNumbers
-resource "port-labs_action" "myAction" {
+resource "port_action" "myAction" {
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-ui-for-action/user-inputs/yaml.md:84]: The user inputs syntax here needs to be updated

Also if there is now support for enum and array, please extend the examples properly

File: docs/customize-pages-dashboards-and-plugins/widgets/async-api.md [hunk 9]
```diff
@@ -55,7 +55,7 @@ To serve the OpenAPI spec from an AWS S3 bucket, please add a CORS policy to the
 <TabItem value="terraform">
 
 ```hcl showLineNumbers
-resource "port-labs_blueprint" "myBlueprint" {
+resource "port_blueprint" "myBlueprint" {
```
Comment (MPTG94) [inline docs/customize-pages-dashboards-and-plugins/widgets/async-api.md:58]: This needs to updated to the new syntax

File: docs/customize-pages-dashboards-and-plugins/widgets/async-api.md [hunk 10]
```diff
@@ -105,7 +105,7 @@ When using the object type, you will have to provide the full JSON AsyncAPI spec
...
```
Comment (MPTG94) [inline docs/customize-pages-dashboards-and-plugins/widgets/async-api.md:108]: This needs to updated to the new syntax

File: docs/customize-pages-dashboards-and-plugins/widgets/embedded-url/embedded-url.md [hunk 11]
```diff
@@ -45,7 +45,7 @@ In order to add an embedded view tab into your specific entity page, all you nee
 <TabItem value="terraform">
 
 ```hcl showLineNumbers
-resource "port-labs_blueprint" "myBlueprint" {
+resource "port_blueprint" "myBlueprint" {
```
Comment (MPTG94) [inline docs/customize-pages-dashboards-and-plugins/widgets/embedded-url/embedded-url.md:48]: This needs to updated to the new syntax

File: docs/customize-pages-dashboards-and-plugins/widgets/open-api.md [hunk 12]
```diff
@@ -55,7 +55,7 @@ To serve the OpenAPI spec from an AWS S3 bucket, please add a CORS policy to the
 <TabItem value="terraform">
 
 ```hcl showLineNumbers
-resource "port-labs_blueprint" "myBlueprint" {
+resource "port_blueprint" "myBlueprint" {
```
Comment (MPTG94) [inline docs/customize-pages-dashboards-and-plugins/widgets/open-api.md:58]: This needs to updated to the new syntax

File: docs/customize-pages-dashboards-and-plugins/widgets/open-api.md [hunk 13]
```diff
@@ -105,7 +105,7 @@ When using the object type, you will have to provide the full JSON OpenAPI spec
...
```
Comment (MPTG94) [inline docs/customize-pages-dashboards-and-plugins/widgets/open-api.md:108]: This needs to updated to the new syntax

**Review by MPTG94 — APPROVED** [review]
Please apply fixes as requested

File: docs/build-your-software-catalog/define-your-data-model/setup-blueprint/properties/array.md [hunk 14]
```diff
@@ -224,17 +248,19 @@ Array validations follow the JSON schema model, refer to the [JSON schema docs](
...
-    # highlight-end
+  properties = {
+    array_props = {
+      "myArrayProp" = {
+        title      = "My array"
+        required   = true
+        # highlight-start
+        min_items  = 0
+        max_items  = 5
+        # highlight-end
```
[the remark this answers is not in the document]
Comment (dvirsegev) [reply docs/build-your-software-catalog/define-your-data-model/setup-blueprint/properties/array.md:261]: Forgot to add , will add it after the merge

File: docs/build-your-software-catalog/sync-data-to-catalog/iac/terraform/terraform.md (hunk 1 above)
Comment (dvirsegev) [reply docs/build-your-software-catalog/sync-data-to-catalog/iac/terraform/terraform.md:319]: many_relations

File: docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/extended-specific-assets.md [hunk 15]
```diff
@@ -200,86 +200,91 @@ resource "port-labs_entity" "gcp_folder_entity" {
...
 
-resource "port-labs_entity" "gcp_project_entity" {
+resource "port_entity" "gcp_project_entity" {
   for_each = {for idx, project in data.google_projects.my_projects.projects : idx => project}
   identifier = each.value.project_id
   title     = each.value.name
-  blueprint = port-labs_blueprint.gcp_project_blueprint.identifier
-  properties {
+  blueprint = port_blueprint.gcp_project_blueprint.identifier
+  properties{
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/gcp/examples/extended-specific-assets.md:254]: This is not according to the new format (also the rest of the occurrences in this file)

File: docs/build-your-software-catalog/sync-data-to-catalog/iac/terraform/terraform.md [hunk 16]
```diff
@@ -122,162 +126,148 @@ resource "port-labs_entity" "myEntity" {
...
 
 </TabItem>
 <TabItem value="number">
 
 ```hcl showLineNumbers
-properties {
-  name = "myNumberProp"
-  value = 7
+properties = {
+  number_prop {
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/iac/terraform/terraform.md]: Need to make sure this is pluralized across the occurrences

File: docs/build-your-software-catalog/sync-data-to-catalog/iac/terraform/terraform.md [hunk 17]
```diff
@@ -122,162 +126,148 @@ resource "port-labs_entity" "myEntity" {
...
 
 </TabItem>
 <TabItem value="url">
 
 ```hcl showLineNumbers
-properties {
-  name = "myUrlProp"
-  value = "https://example.com"
+properties = {
+  string_props = {
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/iac/terraform/terraform.md:187]: Make sure the equals (`=`) sign appears in all necessary places

File: docs/build-your-software-catalog/sync-data-to-catalog/iac/terraform/terraform.md [hunk 18]
```diff
@@ -287,13 +277,15 @@ resource "port-labs_entity" "myEntity" {
...
-    identifier = "myTargetEntityIdentifier"
+    single_relations = {
+    "mySingleRelation" = "myTargetEntityIdentifier"
+    }
   }
 
   relations {
-    name  = "myAdditionalManyRelation"
-    identifiers = ["myAdditionalTargetEntityIdentifier", myAdditionalTargetEntityIdentifier2"]
+    many_relation = {
```
Comment (MPTG94) [inline docs/build-your-software-catalog/sync-data-to-catalog/iac/terraform/terraform.md]: Needs to include an `s` at the end

File: docs/create-self-service-experiences/setup-ui-for-action/setup-ui-for-action.md [hunk 19]
```diff
@@ -198,31 +198,14 @@ The JSON shown above is for a single blueprint action, the actions of a blueprin
...
-    description = "My special user input"
-  }
-
-  trigger = "myActionTrigger"
-
-  invocation_method {
-    type = "myInvocationType"
-  }
+  user_properties = {}
+  trigger = "myActionTrigger"å
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-ui-for-action/setup-ui-for-action.md:207]: Remove this weird character at the end

File: docs/create-self-service-experiences/setup-ui-for-action/user-inputs/text.md [hunk 20]
```diff
@@ -149,7 +177,7 @@ Text validations support the following operators:
 <Tabs groupId="validation-definition" queryString defaultValue="basic" values={[
 {label: "Basic", value: "basic"},
 {label: "Array", value: "array"},
-{label: "Terraform - coming soon", value: "tf"}
+{label: "Terraform", value: "tf"}
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-ui-for-action/user-inputs/text.md:180]: Add the actual code block with the validations

**Review by MPTG94 — APPROVED** [review]
I added a few more small comments

**Pull request merged**